client/modules/driver/reducers/helpers.js

```diff
@@ -27,13 +20,12 @@ export const getGoogleRoute = (waypoints, optimize) => {
       if (status === google.maps.DirectionsStatus.OK) {
         resolve(result)
       } else {
-        reject(result)
+        const error = new Error(status)
```

Comment:
The status should be mapped to a string suitable to show to the user. I think `MAX_WAYPOINTS_EXCEEDED` should show an explanatory message. The rest could maybe just be a default message.